psqlextra/schema.py

```diff
@@ -0,0 +1,214 @@
+    ) -> "PostgresSchema":
+        """Creates a new schema with a random suffix.
+
+        Arguments:
+            prefix:
+                Name to prefix the final name with. The name plus
+                prefix cannot be longer than 63 characters.
+
+            using:
+                Name of the database connection to use.
+        """
+
+        suffix = os.urandom(4).hex()
+        cls._verify_generated_name_length(prefix, suffix)
+
+        return cls.create(f"{prefix}_{suffix}", using=using)
+
+    @classmethod
+    def delete_and_create(
+        cls, name: str, *, cascade: bool = False, using: str = DEFAULT_DB_ALIAS
+    ) -> "PostgresSchema":
+        """Deletes the schema if it exists before re-creating it.
+
+        Arguments:
+            name:
+                Name of the schema to delete+create (max 63 characters).
+
+            cascade:
+                Whether to delete the contents of the schema
+                and anything that references it if it exists.
+
+            using:
+                Optional name of the database connection to use.
+        """
+
+        with transaction.atomic(using=using):
+            cls(name).delete(cascade=cascade, using=using)
+            return cls.create(name, using=using)
+
+    @classmethod
+    def exists(cls, name: str, *, using: str = DEFAULT_DB_ALIAS) -> bool:
+        """Gets whether a schema with the specified name exists.
+
+        Arguments:
+            name:
+                Name of the schema to check of whether it
+                exists.
+
+            using:
+                Optional name of the database connection to use.
+        """
+
+        connection = connections[using]
+
+        with connection.cursor() as cursor:
+            return name in connection.introspection.get_schema_list(cursor)
+
+    def delete(
+        self, *, cascade: bool = False, using: str = DEFAULT_DB_ALIAS
+    ) -> None:
+        """Deletes the schema and optionally deletes the contents of the schema
+        and anything that references it.
+
+        Arguments:
+            cascade:
+                Cascade the delete to the contents of the schema
+                and anything that references it.
+
+                If not set, the schema will refuse to be deleted
+                unless it is empty and there are not remaining
+                references.
+        """
+
+        if self.name == "public":
+            raise SuspiciousOperation(
+                "Pretty sure you are about to make a mistake by trying to drop the 'public' schema. I have stopped you. Thank me later."
+            )
+
+        with connections[using].schema_editor() as schema_editor:
+            schema_editor.delete_schema(self.name, cascade=cascade)
+
+    @classmethod
+    def _verify_generated_name_length(cls, prefix: str, suffix: str) -> None:
+        max_prefix_length = cls.NAME_MAX_LENGTH - len(suffix)
+
+        if len(prefix) > max_prefix_length:
```

Comment:
Assumptions:
1. Prefix len 50
2. Suffix len 13
`max_prefix_length` would be `63 - 13` = `50`
`len(prefix)` = `50` is not `> max_prefix_length` `50`.
The calling function does `{prefix}_{suffix}` which results in 50 **+ 1** + 13 = 64 chars.

Either remove the seperator from the creation functions or consider it here while calculating.

Reply:
Oops, forgot about the separator. Taken into account now.